### PR TITLE
EES-3281 Fix chart reference line position labels

### PR DESCRIPTION
**Pull request by ntsim**
This PR fixes the position labels displayed for chart reference lines in axis configuration. Previously, these would show the underlying filter id/value instead of the human-readable label.

## Related changes

- Updated reference lines to be more visible. Previously, these were essentially invisible to users as they had no clear visual differences from other lines on charts. We've now made reference lines darker, dashed and wider to make them stand out.
- Added Formik to manage state for adding reference lines. This removes the previous hand-rolled form state management and allows us to show validation errors inline with the form inputs.
- Added filtering to remove reference line position options on the major axis. We want to avoid having reference lines for options that are no longer valid (e.g. the user changes the data sets or the axis `groupBy`).
- Added a remediation to prevent stale state being sent to the server when chart builder is saved. Note that this is not a complete solution and it would be better to revisit state management across chart builder to make this better. See [EES-3355](https://dfedigital.atlassian.net/browse/EES-3355) for details.

## Screenshots

Human-readable reference line position labels:

image

More visible reference lines:

image

### CODE REVIEW

**Review by amyb-hiveit — CHANGES_REQUESTED** [review]
Looks good, just a few small comments. Definitely agree about revisiting the chart builder state management.

File: src/explore-education-statistics-admin/src/pages/release/datablocks/components/chart/ChartReferenceLinesConfiguration.tsx [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+  };
+
+  return (
+    <table className="govuk-table">
+      <caption className="govuk-heading-s">Reference lines</caption>
+      <thead>
+        <tr>
+          <th>Position</th>
+          <th>Label</th>
+          <th />
```
[the remark this answers is not in the document]
Comment (ntsim) [reply src/explore-education-statistics-admin/src/pages/release/datablocks/components/chart/ChartReferenceLinesConfiguration.tsx:108]: Good point! The entire thing was just copy and pasted from the parent component. The header is now 'Actions'

File: src/explore-education-statistics-admin/src/pages/release/datablocks/components/chart/ChartReferenceLinesConfiguration.tsx [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+                    ...referenceLine,
+                    label: e.target.value,
+                  });
+                }}
+              />
+            </td>
+            <td>
+              <Button
+                className="govuk-!-margin-bottom-0 dfe-float--right"
+                disabled={!referenceLine.position || !referenceLine.label}
```
[the remark this answers is not in the document]
Comment (ntsim) [reply src/explore-education-statistics-admin/src/pages/release/datablocks/components/chart/ChartReferenceLinesConfiguration.tsx]: I've gone one step further and refactored the add line form to use Formik. Validation errors are now shown inline with the form inputs, and we also show a tooltip and aria-disable the button when the form is invalid.

This should hopefully provide much better feedback than we were previously.

[the remark this answers is not in the document]
Comment (amyb-hiveit) [reply src/explore-education-statistics-admin/src/pages/release/datablocks/components/chart/ChartReferenceLinesConfiguration.tsx]: Nice, thanks.

**Review by amyb-hiveit — APPROVED** [review]

**Pull request merged**